Categorias.java

```diff
@@ -0,0 +1,42 @@
+package casaDoCodigo;
+
+import java.util.Collection;
+import java.util.LinkedList;
+
+public class Categorias extends LinkedList<Categoria> {
+	
+	@Override
+	public void add(int index, Categoria categoria) {
+		for (Categoria c : this) {
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Toda filha de ```Collection``` tem o método ```contains```. Ele já faz essa busca de um objeto igual. No caso de listas, o código é meio parecido com esse :).